client/js/README.md

```diff
@@ -1,31 +1,170 @@
+
+#### `predict`
+
+The `predict` method allows you to call an api endpoint and get a prediction result:
+
+```ts
+import { client } from "@gradio/client";
+
+const app = await client("user/space-name");
+const result = await app.predict("/predict");
+```
+
+`predict` accepts two parameters, `endpoint` and `payload`. It returns a promise that resolves to the prediction result.
+
+##### `endpoint`
+
+This is the endpoint for an api request and is required. The default endpoint for a `gradio.Interface` is `"/predict"`. Explicitly named endpoints have a custom name. The endpoint names can be found on the "View API" page of a space.
+
+```ts
+import { client } from "@gradio/client";
+
+const app = await client("user/space-name");
+const result = await app.predict("/predict");
+```
+
+##### `payload`
+
+The `payload` argument is generally optional but this depends on the API itself. If the API endpoint depends on values being passed in then it is required for the API request to succeed. The data that should be passed in is detailed on the "View API" page of a space.
+
+```ts
+import { client } from "@gradio/client";
+
+const app = await client("user/space-name");
+const result = await app.predict("/predict", [1, "Hello", "friends"]);
 ```
 
-**usage**
+<!-- Types for the different API inputs are provided and the example snippets on the "View API" will provide strong typing for the payload. These types need to be explicitly annotated as there currently no mechanism to automatically infer them due to the varied and dynamic nature of API endpoints. -->
+
+#### `submit`
+
+The `submit` method provides a more flexible way to call an API endpoint, providing you with status updates about the current progress of the prediction as well as supporting more complex endpoint types.
 
 ```ts
 import { client } from "@gradio/client";
 
-const app = client();
+const app = await client("user/space-name");
+const submission = app.submit("/predict", payload);
+```
+
+The `submit` method accepts the same [`endpoint`](#endpoint) and [`payload`](#payload) arguments as `predict`.
+
+The `submit` method does not return a promise and should not be awaited, instead it returns an object with a `on`, `off`, and `cancel` methods.
+
+##### `on`
+
+The `on` method allows you to subscribe to events related to the submitted API request. There are two types of event that can be subscribed to: `"data"` updates and `"status"` updates.
+
+`"data"` updates are issued when the API computes a value, the callback provided as the second argument will be called when such a value is sent to the client. The shape of the data depends on the way the API itself is constructed. This event may fire more than once if that endpoint supports emmitting new values over time.
 
-const prediction = app.predict(endpoint, payload);
+<!-- The type annotations provided by the "View API" page will ensure that the data payload is strongly typed. -->
```

Comment:
remove comment?

Reply:
Yeah i'll remove this and file an issue to add this feature.